examples/e-commerce/src/app/no-results/no-results.component.ts

```diff
@@ -114,6 +118,9 @@ export class NoResults extends BaseWidget {
   public state: NoResultsState = {};
 
   constructor(
+    @Inject(forwardRef(() => NgAisIndex))
+    @Optional()
+    public indexParent: NgAisIndex,
```

Comment:
probably a silly question, but we cannot put them into `BaseWidget`?

```
    @Inject(forwardRef(() => NgAisIndex))
    @Optional()
    public indexParent: NgAisIndex,
```

Reply:
I've tried, and since you're forced to override the constructor (for the widget name in bem), you also are forced to override the constructor with the exact same arity / arguments as BaseWidgets

Reply:
this twitter thread is about it: https://twitter.com/haroenv/status/1199372001502212098 tl;dr: no solution found

Reply:
Ah.. the constructor is overridden :(